modules/pipeline/pipengine/reconciler/taskrun/taskop/start.go

```diff
@@ -159,6 +164,19 @@ func (s *start) TuneTriggers() taskrun.TaskOpTuneTriggers {
 	}
 }
 
+// injectJobID save flink, spark job id after start
+func injectJobID(tr *taskrun.TaskRun, data interface{}) error {
+	if data == nil {
+		return nil
+	}
+	jobID, ok := data.(string)
```

Comment:
You should check task executor type here.
Not all string type data is job id.

Reply:
only dcos return job id, so remove inJectJobID